interlok-core/src/main/java/com/adaptris/core/services/metadata/MetadataAppenderService.java

```diff
@@ -56,6 +62,15 @@ public class MetadataAppenderService extends MetadataServiceImpl {
   @AffectsMetadata
   private String resultKey;
 
+  public static final String DEFAULT_SEPARATOR = "";
+  @NotNull
+  @AutoPopulated
+  @AffectsMetadata
+  @InputFieldDefault("")
+  @Getter
+  @Setter
+  private String separator = DEFAULT_SEPARATOR;
```

Comment:
Maybe we should remove @NoNull and also have a method that will return the default empty string if null.
Or use @InputFieldHint(style = "BLANKABLE") as otherwise in the UI empty value will be consider as null and the user will not be able to save this config.

Reply:
@sebastien-belin-adp 
I made the separator nullable and return the default separator if null. Also made it blankable.